### ISSUE TITLE
§12.21 Boolean expressions wording may be incorrect

### ISSUE THREAD

**Issue by xtqqczze** [dotnet/csharpstandard#367]
Quoting from [draft-v6/standard/expressions.md](https://github.com/dotnet/csharpstandard/blob/draft-v6/standard/expressions.md):

> The controlling conditional expression of the `?:` operator ([§12.15](expressions.md#1215-conditional-operator)) follows the same rules as a *boolean_expression*, but for reasons of operator precedence is classified as a *conditional_or_expression*.

However, since ECMA-334 3rd edition, *conditional_expression* has been defined in terms of *null_coalescing_expression*:

```ANTLR
conditional_expression
    : null_coalescing_expression
    | null_coalescing_expression '?' expression ':' expression
    ;
```

So it is possible the wording in incorrect in either §12.15 or §12.21.

Collating the wording from each edition, one can see the term in §12.21 has gone back and forth:

| ECMA edition | Wording |
| --- | ----------- |
| v1 | *conditional-or-expression* |
| v2 | *conditional-or-expression* |
| v3 | *null-coalescing-expression* |
| v4 | *null-coalescing-expression* |
| v5 | *conditional-or-expression* |
| draft-v6 | *conditional_or_expression* |

**Comment by Nigel-Ecma** [comment]
Well spotted!

The v2 -> v3 change was correct, v3 saw the introduction of *null-coalescing-expression*. The v4 -> v5 change is a regression; it looks like it was unintentional, an accidental consequence of merging in material from another source.

**Comment by Nigel-Ecma** [comment]
@xtqqczze – Would you like to open a PR for this?

**Comment by xtqqczze** [comment]
@Nigel-Ecma I have presented #370 to fix this issue. I haven't marked the PR as a fix for this issue as `standard-v5` and `draft-v7` will require these changes and I don't know how branch ports are managed here,

**Comment by Nigel-Ecma** [comment]
@xtqqczze – #370 approved and in the q for merging. I expect it will find its way into draft-v7 but I’ve handed that on to those who know and will make sure but it does.

**Comment by xtqqczze** [comment]
What about standard-v5, is there an errata list?

**Comment by jskeet** [comment]
@xtqqczze: No, we don't maintain an errata list I'm afraid. (At least not as far as I'm aware.) It's an interesting idea though, and one we'll discuss in our next meeting.

**Comment by Nigel-Ecma** [comment]
Issue resolved by #370

**Comment by xtqqczze** [comment]
@jskeet Did the errata list discussion go anywhere?
